IntegrationTests/UI.IntegrationTests/UITest.cs

```diff
@@ -77,6 +77,29 @@ void HandleApplicationIdle(object sender, EventArgs e)
             }
         }
 
+        public static void RunControl<T>(
+            Func<Form, T> createControl,
+            Func<T, Task> runTestAsync)
+            where T : Control
+        {
+            T control = null;
+            try
+            {
+                RunForm<Form>(
+                    showForm: () =>
+                    {
+                        var form = new Form { Text = $"Test {typeof(T).Name}" };
+                        control = createControl(form);
+                        Application.Run(form);
+                    },
+                    runTestAsync: form => runTestAsync(control));
+            }
+            finally
+            {
+                control?.Dispose();
```

Comment:
❗️ This leaves an uneasy feeling. `RunForm` will dispose of `form` on success and failure paths, and that will dispose of `control` as long as it's a child of `form`. I would rather have an edge case where `control` is not disposed on an exception path than have an invalid call to dispose on the normal path (as well as most exception paths).

Reply:
Changed although a call to `Dispose` cannot be invalid.

Reply:
The implementation of `Dispose` _should_ allow multiple calls, but sometimes does not. Either way, I'm happy with the current approach since we worked so much on making the core runner helper handle disposing in both success and failure cases.